packages/netlify-cms-core/src/actions/search.ts

```diff
@@ -85,84 +67,59 @@ type QueryResponse = {
   query: string;
 };
 
-export function querySuccess(
-  namespace: string,
-  collection: string,
-  searchFields: string[],
-  searchTerm: string,
-  response: QueryResponse,
-) {
+export function querySuccess(namespace: string, hits: EntryValue[]) {
   return {
     type: QUERY_SUCCESS,
     payload: {
       namespace,
-      collection,
-      searchFields,
-      searchTerm,
-      response,
+      hits,
     },
-  };
+  } as const;
 }
 
-export function queryFailure(
-  namespace: string,
-  collection: string,
-  searchFields: string[],
-  searchTerm: string,
-  error: Error,
-) {
+export function queryFailure(error: Error) {
   return {
     type: QUERY_FAILURE,
-    payload: {
-      namespace,
-      collection,
-      searchFields,
-      searchTerm,
-      error,
-    },
-  };
+    payload: { error },
+  } as const;
 }
 
 /*
  * Exported simple Action Creators
  */
 
 export function clearSearch() {
-  return { type: SEARCH_CLEAR };
+  return { type: SEARCH_CLEAR } as const;
 }
 
 /*
  * Exported Thunk Action Creators
  */
 
 // SearchEntries will search for complete entries in all collections.
-export function searchEntries(
-  searchTerm: string,
-  searchCollections: string[] | null = null,
-  page = 0,
-) {
-  return (dispatch: ThunkDispatch<State, {}, AnyAction>, getState: () => State) => {
+export function searchEntries(searchTerm: string, searchCollections: string[], page = 0) {
+  return async (dispatch: ThunkDispatch<State, undefined, AnyAction>, getState: () => State) => {
     const state = getState();
     const { search } = state;
     const backend = currentBackend(state.config);
     const allCollections = searchCollections || state.collections.keySeq().toArray();
     const collections = allCollections.filter(collection =>
-      selectIntegration(state, collection as string, 'search'),
+      selectIntegration(state, collection, 'search'),
     );
-    const integration = selectIntegration(state, collections[0] as string, 'search');
+    const integration = selectIntegration(state, collections[0], 'search');
 
     // avoid duplicate searches
     if (
-      search.get('isFetching') === true &&
-      search.get('term') === searchTerm &&
-      search.get('collections') !== null &&
-      List(allCollections).equals(search.get('collections') as List<string>) &&
+      search.isFetching &&
```

Comment:
do we really need 'isFetching' check here?

Reply:
I think it's useful to refresh the results (you have a succesful search and then search for the same thing).

Reply:
Oh, ok, I understand. You are allowed to search for the same thing twice, but you're guarded from pressing 'enter' multiple times if search hasn't finished yet.